backend/src/stwo/circuit_builder.rs

```diff
@@ -151,18 +177,24 @@ impl<T: FieldElement> FrameworkEval for PowdrEval<T> {
             })
             .collect();
 
-        for id in self
-            .analyzed
-            .identities_with_inlined_intermediate_polynomials()
-        {
+        let intermediate_definitions = self.analyzed.intermediate_definitions();
+        let data = Data {
+            witness_eval: &witness_eval,
+            constant_shifted_eval: &constant_shifted_eval,
+            constant_eval: &constant_eval,
+        };
+        let mut evaluator = ExpressionEvaluator::new_with_custom_expr(
+            &data,
+            &data,
```

Comment:
Just curious why data is passed twice, is it because it's both the global values and the trace values?

Reply:
Yes!